### ISSUE TITLE
Uses of conditional components in ModelicaTest.MultiBody

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1333]
**Reported by jmattsson on 25 Oct 2013 09:46 UTC**
In ModelicaTest.MultiBody.Joints, both UniversalSpherical and SphericalSpherical2 contains the asserts:
```mo
assert(rs1.r_rel[1] < tol2 and rs1.r_rel[2] < tol2 and rs1.r_rel[3] <
  tol2, "Difference in positions between bodies is too much");
assert(rs2.r_rel[1] < tol2 and rs2.r_rel[2] < tol2 and rs2.r_rel[3] <
  tol2, "Difference in positions between bodies is too much");
assert(rs3.r_rel[1] < tol1 and rs3.r_rel[2] < tol1 and rs3.r_rel[3] <
  tol1, "Difference in positions between bodies is too much");
```

This is an error, since r_rel in rs1 is conditional.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1333

**Comment by modelica-trac-importer** [comment]
**Modified by jmattsson on 25 Oct 2013 09:47 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by jmattsson on 21 Nov 2013 08:34 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by jmattsson on 21 Nov 2013 10:11 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 20 Jun 2014 14:09 UTC**
Replying to [ticket:1333 jmattsson]:
> In ModelicaTest.MultiBody.Joints, both UniversalSpherical and SphericalSpherical2 contains the asserts:
> ```mo
> assert(rs1.r_rel[1] < tol2 and rs1.r_rel[2] < tol2 and rs1.r_rel[3] <
>   tol2, "Difference in positions between bodies is too much");
> assert(rs2.r_rel[1] < tol2 and rs2.r_rel[2] < tol2 and rs2.r_rel[3] <
>   tol2, "Difference in positions between bodies is too much");
> assert(rs3.r_rel[1] < tol1 and rs3.r_rel[2] < tol1 and rs3.r_rel[3] <
>   tol1, "Difference in positions between bodies is too much");
> ```
> 
> This is an error, since r_rel in rs1 is conditional.

Changed declaration of rs1, rs2, rs3 to:
```
Modelica.Mechanics.MultiBody.Sensors.RelativeSensor rs1(
    final get_r_rel=true, ...
    ...
```
With "final", the vector is no longer conditional.

Fixed in 6d9ae490ad4d71c93550a5696e84c1a4364be2ea (trunk) and in 6173d2cae19631b74835a2216ce9b6b2267d3970 (maintenance).

**Comment by modelica-trac-importer** [comment]
**Comment by jmattsson on 23 Jun 2014 07:43 UTC**
Declaring the parameter final does not make the component stop being conditional. The current rule in the specification is "A component declared with a condition_attribute can only be modified and/or used in connections."

**Comment by modelica-trac-importer** [comment]
**Comment by otter on 23 Jun 2014 10:00 UTC**
Fixed in 4586fd96b619f6153e489691b6dc0dd400bb8b28 (trunk) and in 4ff0c41e836826dd960328bce3abe267f4bd7d95 (maintenance).
